src/provider/common/Injector.ts

```diff
@@ -0,0 +1,102 @@
+import {Container} from "inversify";
+import {interfaces as inversify} from "inversify/dts/interfaces/interfaces";
+import {Inject} from "./Injectables";
+import {AppDirectory} from "../model/AppDirectory";
+import {IntentHandler} from "../controller/IntentHandler";
+import {ContextHandler} from "../controller/ContextHandler";
+import {Model} from "../model/Model";
+import {SelectorHandler} from "../controller/SelectorHandler";
+import {AsyncInit} from "../controller/AsyncInit";
+import {FinEnvironment} from "../model/Environment";
+import {Environment} from "openfin/_v2/environment/environment";
+
+/**
+ * For each entry in `Inject`, defines the type that will be injected for that key.
+ */
+type Types = {
+    [Inject.APP_DIRECTORY]: AppDirectory,
+    [Inject.CONTEXT_HANDLER]: ContextHandler,
+    [Inject.ENVIRONMENT]: Environment,
+    [Inject.INTENT_HANDLER]: IntentHandler,
+    [Inject.MODEL]: Model,
+    [Inject.SELECTOR]: SelectorHandler
+};
+
+/**
+ * Default injector mappings. Used at startup to initialise injectify.
+ * 
+ * Using a type here will configure injectify to instantiate a class and inject it as a singleton.
+ * Using a value here will inject that instance.
+ */
+const Bindings = {
+    [Inject.APP_DIRECTORY]: AppDirectory,
+    [Inject.CONTEXT_HANDLER]: ContextHandler,
+    [Inject.ENVIRONMENT]: FinEnvironment,
+    [Inject.INTENT_HANDLER]: IntentHandler,
+    [Inject.MODEL]: Model,
+    [Inject.SELECTOR]: SelectorHandler
+};
+
+type Keys = (keyof typeof Inject & keyof typeof Bindings & keyof Types);
+
+/**
+ * Wrapper around inversify that allows more concise injection
+ */
+export class Injector {
+    private static _initialized: Promise<void>;
+
+    private static _container: Container = (() => {
+        const container = new Container();
+        const promises: Promise<unknown>[] = [];
+
+        Object.keys(Bindings).forEach(k => {
+            const key: Keys = k as any;
+
+            if (typeof Bindings[key] === 'function') {
+                container.bind(Inject[key]).to(Bindings[key] as any).inSingletonScope();
```

Comment:
True, since `InjectableMap` creates a maps a set of keys to themselves, there's no difference between the two.

`Inject[key]` _maaaaybe_ reflects the intent a little better - but yeah, no difference...